### DISCUSSION TITLE
New device discovered?

### DISCUSSION THREAD

**Discussion by rcdailey** [krahabb/meross_lan#182] [Q&A]
I got a new device discovered notification today. What changed in the integration?

![Screenshot_20220323-171035](https://user-images.githubusercontent.com/1768054/159804750-28061959-89c1-403c-a175-c2f0121b04c3.png)

**Comment by CTJohnK** [comment]
I have only seen that when I either reset a device and added back to the Meross app or I added a new device.  I presume neither are the case for you?

I have not upgraded to the most recent Meross LAN version, but is there a chance your device's IP address changed?

**Comment by krahabb** [comment]
Like @CTJohnK said, this is likely due to a 'sudden' IP change on the device. Even though meross_lan should be able to match the mac address to an existing one and then discard the discovery, this flow has never been tested in details so it might fail somehow.
What happens if you try to configure it? 
Also, if an already configured device IP changed, that device should fall offline unless your setup is working on MQTT too (this would keep  working since MQTT communication doesn't care about IP address and meross_lan is able to anyway retrieve the updated IP from the payloads exchanged over MQTT)

**Comment by wsw70** [comment]
I have the same case. The weird thing is that this is different from other discovered devices. The ones I am used to are presented with an ID and the configuration is just a choice of the room.

The new one is like that (when choosing to configure):

![image](https://user-images.githubusercontent.com/1345886/183309471-7653bbb2-192b-4089-8d96-f859f6253103.png)

It is so different from the others that I thought it was a different integration that somehow appeared :)

All my devices work so I have no idea what this is. It advertises itself as a Meross Smart Plug though.

![image](https://user-images.githubusercontent.com/1345886/183309610-200e1f50-2cfe-4a53-8ead-20bfba5d9be4.png)

**Comment by krahabb** [comment]
This is likely discovered through dhcp.
When a new dhcp lease is announced meross_lan gets notified by HA and tries to see if it matches any of the already configured one. If not you'll be prompted with the discovery dialog.
In order to detect if the device is already configured, meross_lan tries a mixed approach by checking against known macs and UUID to see if the new one is already registered (and then discard the dhcp announcement). If not, this is what happens. keep in mind that, in order to see if the device UUID is already registered you likely have to enter the device key so it might happen meross_lan gets fooled by a new discovery while the device is not strictly new...
Let me know if it appears this is the case!
